src/views/settings/filters/parts/modify-filter/modify-filter-modal.tsx

```diff
@@ -141,8 +143,12 @@ const ModifyFilterModal: FC<ComponentProps> = ({
 	const incomingFiltersCopy = useMemo(() => incomingFilters?.slice(), [incomingFilters]);
 
 	const toggleCheckBox = useCallback(() => {
-		setDontProcessAddFilters(!dontProcessAddFilters);
-	}, [dontProcessAddFilters]);
+		setDontProcessAddFilters((prev) => !prev);
+	}, []);
+
+	useMemo(() => {
```

Comment:
Whenever we change the value of the dontProcessAddFilters state and it will re-render all the components as well as calling the useEffect so states value will be change to the old one again. We are planning to refactor the filters so at them moment added this solution.